### PR TITLE
Operator hub onboarding

### PR DESCRIPTION
**Pull request by anishasthana**
I will be submitting this PR (or a very very similar one) to the community operatorhub repo

### CODE REVIEW

**Review by KPostOffice — COMMENTED** [review]
Is everything from the "Create Operator Bundle" commit auto generated?

File: Dockerfile [hunk 1]
```diff
@@ -21,5 +21,5 @@ WORKDIR /
 COPY --from=builder /workspace/manager .
 COPY config/internal config/internal
 
-
```
Comment (KPostOffice) [inline Dockerfile:24]: Why is this user `1000` should it use user 65532 here as it did before

File: Makefile [hunk 2]
```diff
@@ -101,9 +101,6 @@ fmt: ## Run go fmt against code.
 vet: ## Run go vet against code.
 	go vet ./...
 
-.PHONY: test
-test: manifests generate fmt vet envtest ## Run tests.
```
Comment (KPostOffice) [inline Makefile:105]: How should I run tests locally after this change?

Comment (anishasthana) [reply Makefile:105]: We still have the `test-unit`directive.

**Comment by anishasthana** [comment]
>Is everything from the "Create Operator Bundle" commit auto generated?
Mix of auto-generated and prompt-guided inputs.

**Review by KPostOffice — COMMENTED** [review]
/lgtm

**Comment by anishasthana** [comment]
Looks like one of the roles is not being applied correctly.

File: config/manifests/bases/codeflare-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    alm-examples: '[]'
+    capabilities: Basic Install
+  name: codeflare-operator.v0.0.0
+  namespace: placeholder
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+    - description: InstaScale is the Schema for the instascales API
+      displayName: Insta Scale
```
Comment (Maxusmusti) [inline config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:14]: Is there a reason for the space here? (InstaScale)

File: config/manifests/bases/codeflare-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    alm-examples: '[]'
+    capabilities: Basic Install
+  name: codeflare-operator.v0.0.0
```
Comment (Maxusmusti) [inline config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:7]: Should the version be v0.0.0, or should it start at v0.0.1 (like how we did our images)?

File: config/manifests/bases/codeflare-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (anishasthana) [reply config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:14]: Ah, the reason would be that the underlying display name in the CRD is wrong. (this was auto generated from those).

File: config/manifests/bases/codeflare-operator.clusterserviceversion.yaml (hunk 4 above)
Comment (anishasthana) [reply config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:7]: Ha, 0.0.1 would be right. Will fix it

File: config/manifests/bases/codeflare-operator.clusterserviceversion.yaml (hunk 3 above)
Comment (anishasthana) [reply config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:14]: My vote would be a follow up PR later to fix them up

Comment (anishasthana) [reply config/manifests/bases/codeflare-operator.clusterserviceversion.yaml:14]: Opened #59

File: bundle/manifests/codeflare-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,549 @@
...
+          },
+          "spec": {}
+        }
+      ]
+    capabilities: Basic Install
+    operators.operatorframework.io/builder: operator-sdk-v1.24.1
+    operators.operatorframework.io/project_layout: go.kubebuilder.io/v3
+    operatorframework.io/suggested-namespace: openshift-operators
+  name: codeflare-operator.v0.0.1
+  namespace: placeholder
```
Comment (tedhtchang) [inline bundle/manifests/codeflare-operator.clusterserviceversion.yaml:36]: Perhaps add the repository and containerImage to the metadata. For example:

```
      repository: https://github.com/project-codeflare/codeflare-operator.git
      containerImage: <img registry >/<namspace>/<repo name>:v0.0.1
```

Comment (anishasthana) [reply bundle/manifests/codeflare-operator.clusterserviceversion.yaml:36]: I'm not seeing it under the csv spec. Are you saying additional links, like so? The below code snippet is basically from the bottom of the file

```
  links:
  - name: Codeflare Operator
    url: https://codeflare.dev/
  - name: Container image
    url: quay.io/project-codeflare/codeflare-operator:0.0.1
  - repository: https://github.com/project-codeflare/codeflare-operator.git
  maintainers:
  - email: aasthana@redhat.com
    name: Anish Asthana
  - email: kpostlet@redhat.com
    name: Kevin Postlethwait
  - email: meyceoz@redhat.com
    name: Mustafa Eyceoz
  maturity: alpha
  provider:
    name: CodeFlare
    url: https://github.com/project-codeflare
  version: 0.0.1
```

I"m not necessarily seeing the value of additional links as:
1) The container Image is already specified in the CSV deployment spec
2) The repository is linked under "provider"

Comment (anishasthana) [reply bundle/manifests/codeflare-operator.clusterserviceversion.yaml:39]: Ted explained to me offline what he meant, I've made the updates.

**Review by Maxusmusti — APPROVED** [review]
/lgtm

**Pull request merged**